nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/FetchAzureBlobStorage_v12.java

```diff
@@ -109,11 +100,11 @@ public class FetchAzureBlobStorage_v12 extends AbstractAzureBlobProcessor_v12 {
             .build();
 
     private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(
-            STORAGE_CREDENTIALS_SERVICE,
-            CONTAINER,
-            BLOB_NAME,
-            RANGE_START,
-            RANGE_LENGTH
+        CONTAINER,
+        BLOB_NAME,
+        STORAGE_CREDENTIALS_SERVICE,
+        RANGE_START,
+        RANGE_LENGTH
```

Comment:
@turcsanyip that makes sense from a purely organizational "like things should go together" perspective. When a user goes to configure a processor, though, there are certain things that the user will definitely expect to configure. The first thing that comes to mind is "What data do I want to fetch?" That's configured via the container and blob name. Next is authentication. So that should be the first thing presented to the user. We want to present the most important configuration options first. After that, it makes sense to group things with similar importance by categories. So so RANGE_START should definitely be located with RANGE_LENGTH. But it doesn't really make sense put those before credentials because they are far less important (in terms of what the user will want to configure). And it doesn't make sense to me to configure the Container, then the Credentials, then the Blob, because the Container and Blob together represent "what's the data that I want?"

Reply:
@markap14 indeed, moving `STORAGE_CREDENTIALS_SERVICE` down further, I suggested in the comment, was not a good idea.
Actually, when it was placed at the first position originally, I also followed the "What data do I want to fetch?" approach. In Azure the container name is not globally unique (unlike bucket name on S3) but the container exists within a Storage Account. So, for identifying the data, the user needs: Storage Account > Container > Blob. `STORAGE_CREDENTIALS_SERVICE` contains the `Storage Account Name`, that's why it was placed before `Container` (and also because the credential is a property to be configured always).